geps/gep-713/index.md

```diff
@@ -3,1155 +3,1172 @@
-  // or semantically invalid.
-  PolicyReasonInvalid PolicyConditionReason = "Invalid"
+In the context of traffic networking, for example, often the question asked by users is *“What happens when a network request X comes in?”*. There is an implicit expectation that a set of Kubernetes resources suffices to represent all the rules for a given workload to be activated and thus process request X, and often that is the case. For more complex cases however (multiple personas, application concerns separated into dedicated resource kinds, interaction between groups of users, etc), real life can get more complicated than a simple `kubectl get x`, or at least additional steps must be automated to encompass complexity into what can be achieved with a single declarative object.
 
-  // PolicyReasonTargetNotFound is used with the "Accepted" condition when the policy is attached to
-  // an invalid target resource
-  PolicyReasonTargetNotFound PolicyConditionReason = "TargetNotFound"
-)
-```
+With that in mind, a possible solution for the discoverability problem may involve designing tools (e.g. CLI tools/plugins), new CRDs, etc that let users ask questions in terms of the real life problems they have to deal with on a daily basis, rather than shaped by the underlying technologies used in the process. For instance, a simple Kubernetes object that is used to declare the rules to process a HTTP request cannot have its status reported simply as Ready/Not ready. By being a complex object composed of multiple routing rules, potentially affected by specifications declared from other objects as well, its status MUST account for that complexity and be structured in such a way that informs the owner with respect to each possible case, whether the ones induced by the internal specification declared by the object itself or its external relationships.
+
+In other words, the discoverability problem exists and must be addressed in light of the complexity associated with the topology of nested contexts induced by a set of hierarchically related resources. One should always have that topology in mind while asking questions regarding the behavior of a given resource, because just like a routing object (e.g. HTTPRoute) does not exist independently from its parent contexts (e.g. Gateways) or its children (e.g. Backends), any resource in focus may be just a part of a whole.
+
+### Status reporting
+
+#### Metaresource status
+
+Metaresource CRDs MUST define a status stanza that allows for reporting the status of the metaresource with respect to each scope the resource may apply.
+
+The basic status conditions are:
+
+* **Accepted**: the metaresource passed both syntactic validation by the API server and semantic validation enforced by the controller, such as whether the target objects exist.
+* **Enforced**: the metaresource’s spec is guaranteed to be fully enforced, to the extent of what the controller can ensure.
```

Comment:
This feels an awful lot like the previous "Ready" condition on Gateways - near impossible to implement in reality. I'd rather aim for something like "Programmed" here.

Reply:
We were also proposing `PartiallyEnforced`, which is an intermediary state between `Enforced` and `Overridden`.

However, `PartiallyProgrammed` sounds weird to me. It seems to suggest the controller did not finish programming the policy, which is not what we want to convey.

So, if we change `Enforced` to `Programmed`, would you say `PartiallyEnforced` should then become `PartiallyOverridden`?

cc @robscott